### ISSUE TITLE
Store TAG selection based on the name+type of item instead of individual item ID. 

### ISSUE THREAD

**Issue by Wh1rledPeas** [DestinyItemManager/DIM#2833]
I find it frustrating when I get a higher level drop of an item I've already flagged as a favorite (for example) and if I forget to tag the new one, I lose the tag.

It would be nice if the new item showed the same tag.

Of course the infuse and discard tags might not apply, so this might not be preferable to everyone.

I'm not sure how other people use the discard flag, but I use it in the item compare screen, so I can flag items in my vault that are lower in light, so I can go find them easier.

**Comment by bhollis** [comment]
This is definitely something we've discussed in the D2 world. I'm not sure how useful this'd be for when we have random rolls again. There's also been discussion of having the ability to set up auto-tagging of new items using various rules.

**Comment by Wh1rledPeas** [comment]
A tagging rule system would be wonderful, but I assume much more complex. Especially since it would need the ability to compare items. For example when you have one item that is lower in power than one of the dupes, but is a masterworks and it should be marked keep, while the other should be marked infuse. Or in the case of random rolls, needing to examine the various properties. 

Perhaps it would be enough to automatically copy tags from a duplicate item when a new item is detected. If a newitem() event isn't thrown then perhaps a scan for any items without a tag and any available duplicate with a tag. 

Of course that doesn't take into account the case of multiple duplicates that may be tagged differently. 

I didn't think about the random rolls issue. You have a good point.

My only issue is having to remember to copy the tag to new items.

**Comment by Wh1rledPeas** [comment]
Perhaps you should implement a custom DIMscript language that allows users to create their own automation? 😎

**Comment by bhollis** [comment]
It's definitely annoying, but I can think of a lot of edge cases for auto tagging. For example, if you have a masterwork item tagged favorite and you get a non-masterwork version, is that also a favorite?

**Comment by Wh1rledPeas** [comment]
I see where that could create conflicts. Perhaps you could only copy tags to the untagged item when the user uses the infuse button? Or perhaps prompt? That would be a use case when the user is conscious of the fact that they're either going to infuse or discard the duplicate.

But... The infuse dialog doesn't layout properly on my phone, so I can never use it. So that option wouldn't help mobile phone users.

**Comment by bhollis** [comment]
That’s a bunmer. We have a lot to do to make the older stuff work well on mobile.

**Comment by bhollis** [comment]
If you can post a screenshot of the infusion fuel finder and let me know what kind of phone you have, maybe I can fix it up a little bit!

**Comment by Wh1rledPeas** [comment]
I'm on Github mobile and I don't see a way to attach, so here's a link...

http://imgur.com/a/pt7oW6B

I have a few ideas for the infuse tool and mobile interface. I should probably post them as individual issues so you can track them individually. They're a little off topic for this issue.

**Comment by bhollis** [comment]
That'd be awesome, thanks!

I can see how the screen isn't working out as is - in the future we're hoping to introduce a scrollable panel for mobile that avoids this kind of thing where everything gets squished.

**Comment by Wh1rledPeas** [comment]
Honestly, I'd rather not have to scroll. That's more of a hastle. I think the mobile interface can be tweaked enough to supply a great interface without a lot of scrolling. 

I use Ishtar Commander most of the time for maximizing power level and quickly transferring items. Simply because the interface is so much snappier. It's by far, the best mobile interface to date. (in my opinion)

I use DIM when it comes to comparing and finding items. Now that Ishtar doesn't provide load outs or configuration preferences anymore, I'm starting to depend on DIM more though. 

As a side note, the check boxes used to wrap properly. I just noticed when I took these screen shots, that they aren't wrapping properly now.
